src/libs/Clipboard/index.js

```diff
@@ -6,6 +6,23 @@ import * as Browser from '../Browser';
 
 const canSetHtml = () => lodashGet(navigator, 'clipboard.write');
 
+/**
+ * Check if an HTMLElement is text input or text area
+ * @param {HTMLElement} el
+ * @returns {Boolean}
+ */
+const isTextElement = (el) => {
+    if (el instanceof HTMLInputElement) {
+        // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types
+        if (/|text|email|number|password|search|tel|url/.test(el.type || '')) {
```

Comment:
Do we need this part? I found that selection still works after removing this check, can you confirm why we need this check?

Reply:
OK so I played around with this, and since the clipboard is only used in the chat page, we only need to detect if the element is `HTMLTextAreaElement` which is what the composer uses. So this is how the change would look like:

```diff
diff --git a/src/libs/Clipboard/index.js b/src/libs/Clipboard/index.js
index 14399ebde2..702cc5dd01 100644
--- a/src/libs/Clipboard/index.js
+++ b/src/libs/Clipboard/index.js
@@ -6,22 +6,7 @@ import * as Browser from '../Browser';
 
 const canSetHtml = () => lodashGet(navigator, 'clipboard.write');
 
-/**
- * Check if an HTMLElement is text input or text area
- * @param {HTMLElement} el
- * @returns {Boolean}
- */
-const isTextElement = (el) => {
-    if (el instanceof HTMLInputElement) {
-        // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types
-        if (/|text|email|number|password|search|tel|url/.test(el.type || '')) {
-            return true;
-        }
-    }
-    if (el instanceof HTMLTextAreaElement) { return true; }
-
-    return false;
-};
+const isComposer = el => el instanceof HTMLTextAreaElement;
 
 /**
  * Deprecated method to write the content as HTML to clipboard.
@@ -47,9 +32,9 @@ function setHTMLSync(html, text) {
 
     const selection = window.getSelection();
     let originalSelection = null;
-    const firstAnchorChild = selection.anchorNode && selection.anchorNode.firstChild;
+    const firstAnchorChild = selection.anchorNode.firstChild;
 
-    if (firstAnchorChild && isTextElement(firstAnchorChild)) {
+    if (isComposer(firstAnchorChild)) {
         originalSelection = {
             start: firstAnchorChild.selectionStart,
             end: firstAnchorChild.selectionEnd,
@@ -78,7 +63,7 @@ function setHTMLSync(html, text) {
 
     selection.removeAllRanges();
 
-    if (firstAnchorChild && isTextElement(firstAnchorChild)) {
+    if (isComposer(firstAnchorChild)) {
         firstAnchorChild.setSelectionRange(originalSelection.start, originalSelection.end, originalSelection.direction);
     } else {
         selection.setBaseAndExtent(originalSelection.anchorNode, originalSelection.anchorOffset, originalSelection.focusNode, originalSelection.focusOffset);

 
```

Reply:
We are trying to use `setSelectionRange`, according to the [WHATWG forms spec](https://html.spec.whatwg.org/multipage/forms.html#concept-input-apply) selectionStart, selectionEnd properties, and setSelectionRange method apply only to inputs of types text, search, URL, tel and password.  (I should remove `email` and `number`)

This is what happened if an element is not supported 
<img width="1436" alt="Screen Shot 2023-01-24 at 22 05 45" src="https://user-images.githubusercontent.com/11847279/214331308-a0e1f61d-cd11-4d07-b833-71e7148c81da.png">

Reply:
@youssef-lr I thought there is a chance it will be used for textinput (e.g. search bar), yeah currently it only is used for textarea only. Should I change it to `el instanceof HTMLTextAreaElement` (without a separate function)?

Reply:
Yes, check the [diff](https://github.com/Expensify/App/pull/14506#discussion_r1085429567) I commented above which simplifies code further.

Reply:
@youssef-lr, I've updated the code and a null check in the below part will still be required, unless we will accommodate the right-click issue. I still believe that the modal issue is upstream since there is no suspicious `onmousedown` or `onmouseup` that will make the text deselected on safari only.


```
    const firstAnchorChild = selection.anchorNode && selection.anchorNode.firstChild;
```